### PR TITLE
SUS-1508: Introduce limit parameter for domains query

### PR DESCRIPTION
**Pull request by mszabo-wikia**
https://wikia-inc.atlassian.net/browse/SUS-1508

### CODE REVIEW

**Review by macbre — APPROVED** [review]
`city_id` are obviously not continuous values (as we remove unactive wikis), but still :+1:  :)

File: extensions/wikia/WikiaApi/WikiaApiQueryDomains.php [hunk 1]
```diff
@@ -59,6 +59,23 @@ public function execute() {
...
+				} else {
+					$maxLimit = $settings['limit'][ApiBase::PARAM_MAX];
+				}
+
+				$requested = 1 + $to - $from;
+				if ( $requested > $maxLimit ) {
+					$this->setWarning( "Count of results must not be over $maxLimit (set to $requested)" );
+				}
+
+				$limit = min( $requested, $maxLimit );
```
Comment (ghost) [inline extensions/wikia/WikiaApi/WikiaApiQueryDomains.php:77]: Do we want `$limit` to be set only when `$from` and `$to` are not empty? What if any of those is not set? No limit at all?

File: extensions/wikia/WikiaApi/WikiaApiQueryDomains.php [hunk 2]
```diff
@@ -85,6 +102,7 @@ public function execute() {
 		} else {
 			$this->addFields(array('city_id', 'city_url', 'city_lang'));
 			$this->addOption( "ORDER BY ", "city_id" );
+			$this->addOption( 'LIMIT', $limit );
```
Comment (ghost) [inline extensions/wikia/WikiaApi/WikiaApiQueryDomains.php:105]: Possible use of undeclared variable here and MySQL error.

**Review by ghost — CHANGES_REQUESTED** [review]

File: extensions/wikia/WikiaApi/WikiaApiQueryDomains.php (hunk 2 above)
Comment (mszabo-wikia) [reply extensions/wikia/WikiaApi/WikiaApiQueryDomains.php:105]: If the variable is not provided by user in the URL, MW API will set it to the default value declared on https://github.com/Wikia/app/pull/12627/files#diff-7f7f6f25d0e2d3ff0f4770cea8e296dbR165 . So without a provided parameter, limit will be 10.

Comment (ghost) [reply extensions/wikia/WikiaApi/WikiaApiQueryDomains.php:105]: Thanks!

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
Tested and accepted.

**Pull request merged**